src/Domain/Creators/BinaryCompactObject.cpp

```diff
@@ -346,6 +463,35 @@ BinaryCompactObject::initial_refinement_levels() const noexcept {
 std::unordered_map<std::string,
                    std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
 BinaryCompactObject::functions_of_time() const noexcept {
-  return {};
+  std::unordered_map<std::string,
+                     std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
+      result{};
+  if (!enable_time_dependence_) {
+    return result;
+  }
+
+  const double initial_expiration_time =
+      initial_expiration_delta_t_ ? initial_time_ + *initial_expiration_delta_t_
+                                  : std::numeric_limits<double>::max();
```

Comment:
I just wanted the functions of time to never expire but also not throw an FPE, so I thought max() was the way to go. I could test if infinity() also got the same result if you like

Reply:
I'll change this to infinity(), because I realize that i) comparing a finite number to infinity() should not throw a floating point exception, and ii) doing math on the expiration delta t (which does throw an exception) shouldn't happen, so might as well make sure